pkg/reconciler/ingress/status.go

```diff
@@ -277,63 +370,65 @@ func (m *StatusProber) processWorkItem() bool {
 	if err != nil || !ok {
 		// In case of error, enqueue for retry
 		m.workQueue.AddRateLimited(obj)
-		m.logger.Errorf("Probing of %s failed: ready: %t, error: %v (depth: %d)", item.podIP, ok, err, m.workQueue.Len())
+		m.logger.Errorf("Probing of %s failed, IP: %s, ready: %t, error: %v (depth: %d)", item.url, item.podIP, ok, err, m.workQueue.Len())
 	} else {
-		// In case of success, update the state
-		if atomic.AddInt32(&item.pendingCount, -1) == 0 {
-			m.readyCallback(item.virtualService)
-		}
+		m.updateStates(item.ingressState, item.podState)
 	}
 	return true
 }
 
-// listVirtualServicePodIPs lists the IP addresses of the Envoy pods impacted by the provided VirtualService.
-func (m *StatusProber) listVirtualServicePodIPs(vs *v1alpha3.VirtualService) ([]string, error) {
-	var podIPs []string
-	for _, name := range vs.Spec.Gateways {
-		// Only the ingress gateways are probed
-		if name == "mesh" {
-			continue
-		}
+func (m *StatusProber) updateStates(ingressState *ingressState, podState *podState) {
+	if atomic.AddInt32(&podState.successCount, 1) == 1 {
```

Comment:
`>=1`, perhaps?

Reply:
No, as the comment suggest, we want to execute the branch only once when the first probing call succeeds. All subsequent probing calls are no-op.
If it were to be executed more than once, it would screw up the count of pending pods.